src/main/java/com/upsolver/datasources/jdbc/querybuilders/SqlServerQueryDialect.java

```diff
@@ -1,9 +1,24 @@
 package com.upsolver.datasources.jdbc.querybuilders;
 
+import com.upsolver.datasources.jdbc.metadata.SimpleSqlType;
+import microsoft.sql.Types;
+
+import java.lang.reflect.Field;
 import java.sql.Connection;
+import java.sql.JDBCType;
 import java.sql.SQLException;
+import java.sql.SQLType;
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+import java.util.Optional;
+import java.util.stream.Collectors;
 
 public class SqlServerQueryDialect extends DefaultQueryDialect {
+    private static final Map<Integer, SQLType> sqlServerTypes = Arrays.stream(Types.class.getFields())
+            .map(f -> new SimpleSqlType(f.getName(), "microsoft", getIntValue(f))).collect(Collectors.toMap(SimpleSqlType::getVendorTypeNumber, t -> t));
+    private static final Collection<SQLType> sqlServerTimeTypes = sqlServerTypes.values().stream().filter(f -> f.getName().contains("DATE") || f.getName().contains("TIME")).collect(Collectors.toSet());
```

Comment:
Seems too loose. A direct list is safer

Reply:
Probably, but more verbose.

Reply:
I prefer the more verbose method if there is a potential for an issue otherwise later.